Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/BaseClasses/Effectiveness.mo

```diff
@@ -0,0 +1,156 @@
+    "Supply air volumetric flow rate"
+    annotation (Placement(transformation(extent={{-140,60},{-100,100}})));
+  Modelica.Blocks.Interfaces.RealInput VExh_flow( final unit="m3/s")
+    "Exhaust air volumetric flow rate"
+    annotation (Placement(transformation(extent={{-140,20},{-100,60}})));
+  Modelica.Blocks.Interfaces.RealInput wheSpe(final unit="1")
+    "Wheel speed ratio"
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}})));
+  Modelica.Blocks.Interfaces.RealOutput epsSen(final unit="1")
+    "Sensible heat exchanger effectiveness"
+    annotation (Placement(transformation(extent={{100,30},{120,50}})));
+  Modelica.Blocks.Interfaces.RealOutput epsLat(final unit="1")
+    "Latent heat exchanger effectiveness"
+    annotation (Placement(transformation(extent={{100,-50},{120,-30}})));
+
+protected
+  Real rat
+    "Ratio of the average operating volumetric air flow rate to the nominal supply air flow rate";
+  Real epsSenPL
+    "Part load (75%) sensible heat exchanger effectiveness used for calculation";
+  Real epsSen_nominal
+    "Nominal sensible heat exchanger effectiveness used for calculation";
+  Real epsLatPL
+    "Part load (75%) latent heat exchanger effectiveness used for calculation";
+  Real epsLat_nominal
+    "Nominal latent heat exchanger effectiveness used for calculation";
+
+equation
+  // check if the air flows are too unbalanced.
+  assert(noEvent(VSup_flow - 2*VExh_flow < 0) or noEvent(VExh_flow - 2*VSup_flow < 0),
+    "*** Warning in " + getInstanceName() + ": The ratio of the supply flow rate to the exhaust flow rate should be in the range of [0.5, 2].",
+    level=AssertionLevel.warning);
+  // calculate the average volumetric air flow and flow rate ratio.
+  rat = (VSup_flow + VExh_flow)/2/VSup_flow_nominal;
+  // check if the extrapolation goes too far.
+  assert(noEvent(rat > 0.5) and noEvent(rat < 1.3),
+    "*** Warning in " + getInstanceName() + ": The ratio of the operating flow rate to the nominal supply flow rate should be in the range of [0.5, 1.3].",
+    level=AssertionLevel.warning);
+  // switch between cooling and heating modes based on the difference between the supply air temperature and the exhaust air temperature.
+  epsSenPL = Buildings.Utilities.Math.Functions.regStep(TSup-TExh, epsSenCooPL, epsSenHeaPL, 1e-5);
+  epsSen_nominal = Buildings.Utilities.Math.Functions.regStep(TSup-TExh, epsSenCoo_nominal, epsSenHea_nominal, 1e-5);
+  epsLatPL = Buildings.Utilities.Math.Functions.regStep(TSup-TExh, epsLatCooPL, epsLatHeaPL, 1e-5);
+  epsLat_nominal = Buildings.Utilities.Math.Functions.regStep(TSup-TExh, epsLatCoo_nominal, epsLatHea_nominal, 1e-5);
+  // calculate effectiveness.
+  epsSen =wheSpe*(epsSenPL + (epsSen_nominal - epsSenPL)*(rat - 0.75)/0.25);
+  epsLat =wheSpe*(epsLatPL + (epsLat_nominal - epsLatPL)*(rat - 0.75)/0.25);
+  assert(epsSen > 0 and epsSen < 1,
+    "Insensed value for the sensible heat exchanger effectiveness",
```

Comment:
what does the `Insensed` mean? Can you improve this message like `*** Error: abcde`. In the message, you may add information of how to avoid the error.

Reply:
Thanks for your comments, the messages are updated.